### ISSUE TITLE
Database ref from trigger throws error on firebase function emulator

### ISSUE THREAD

**Issue by carrickpaul** [firebase/firebase-functions#726]
### Environment info

**firebase-tools:**
8.4.3

**Platform:**
Windows 10

### Test case

```
import * as functions from 'firebase-functions'

exports.onDeviceOnlineChanged = functions.database.ref('/status/{deviceId}')
  .onWrite(async (change, context) => {
    const ref = change.before.ref.toString()
    // do something with the reference
    return ref
  })
```

###  Steps to reproduce
make a change to that node on the realtime database to initiate the function

###  Expected behavior
No error, this works on the server, but not on the emulator.

Calling the same code directly from the app runs without issues. The call to the  `change.before.ref` causes the error only when the function is initiated from a trigger.

Calling using the initiated  `admin.database().ref('/status/${deviceId}')` works normally as in the snippet below
```
exports.onDeviceChanged = functions.https.onCall(async (data, context) => {
  const deviceId = data.deviceId
  const status = admin.database().ref(`/status/${deviceId}`)

  const deviceFirestoreRef = admin.firestore().collection('devices').doc(deviceId)

  const upd = { online: status.toJSON() }
  return deviceFirestoreRef.update(upd);
})
```

### Actual behavior
throws the error:
@firebase/database: FIREBASE FATAL ERROR: Cannot parse Firebase url. Please use https://<YOUR FIREBASE>.firebaseio.com

```
!  functions: Error: FIREBASE FATAL ERROR: Cannot parse Firebase url. Please use https://<YOUR FIREBASE>.firebaseio.com 
    at fatal (C:\dev\xyz\functions\node_modules\@firebase\database\dist\index.node.cjs.js:341:11)
    at parseRepoInfo (C:\dev\xyz\functions\node_modules\@firebase\database\dist\index.node.cjs.js:1295:9)
    at RepoManager.databaseFromApp (C:\dev\xyz\functions\node_modules\@firebase\database\dist\index.node.cjs.js:14919:25)
    at Object.initStandalone (C:\dev\xyz\functions\node_modules\@firebase\database\dist\index.node.cjs.js:15303:45)
    at DatabaseService.getDatabase (C:\dev\xyz\functions\node_modules\firebase-admin\lib\database\database.js:66:23)
    at FirebaseApp.database (C:\dev\xyz\functions\node_modules\firebase-admin\lib\firebase-app.js:231:24)
    at DataSnapshot.get ref [as ref] (C:\dev\xyz\functions\node_modules\firebase-functions\lib\providers\database.js:279:34)
    at C:\dev\xyz\functions\lib\presense.js:7:31
    at cloudFunction (C:\dev\xyz\functions\node_modules\firebase-functions\lib\cloud-functions.js:132:23)
    at C:\Users\P\AppData\Roaming\npm\node_modules\firebase-tools\lib\emulator\functionsEmulatorRuntime.js:563:20
!  Your function was killed because it raised an unhandled error.
```

**Comment by samtstern** [comment]
Ok @carrickpaul I was able to track down this issue.  Fixing it would require changes in both the Functions emulator and the Firebase Functions SDK:

https://github.com/firebase/firebase-functions/pull/727
https://github.com/firebase/firebase-tools/pull/2434

**Comment by carrickpaul** [comment]
thanks @samtstern

**Comment by samtstern** [comment]
The fix has been merged in to https://github.com/firebase/firebase-tools/pull/2434 so this will be fixed completely in the next release of `firebase-tools` (as long as it's used with `firebase-functions` 3.8.0 or higher)

**Comment by prescottprue** [comment]
Thanks a bunch for the fixes posted @samtstern 

Even after updating both firebase-tools, and firebase-functions to the noted versions, I still get this error, but now from `@firebase/database` instead (^0.6.0 is pointed to, but still experienced it with a fresh install which pulled 0.6.8 of @firebase/database). This is also being experienced by multiple devs on my team

Let me know if you need any more info or would like me to open a new issue

**Comment by ElGoorf** [comment]
I noticed the same, but a release for firebase-tools hasn't been released
since this update, so from my understanding we just need to wait a while
longer

**Comment by prescottprue** [comment]
@ElGoorf It appears you are correct - v8.6.0 of firebase-tools was just released a few minutes go and it appears that fixed it. Got confused since I didn't see 2434 mentioned anywhere in the changelog.

Thanks @samtstern

**Comment by samtstern** [comment]
@prescottprue the fix in the CL was actually only released a few hours ago in version 8.6.0 ... before that it was merged to master but not released.  Can you try with that version?

**Comment by samtstern** [comment]
Oops didn't see your last comment when replying.  Glad it works now!
